Added local configuration for robot testsuites management.

namsonx opened the pull request:
Hello Thomas,

     I have just created new pull-request for local configuration feature.
     The additional test cases for acceptance test are also added.
     The documentation also updated regarding to this new feature.

Thank you,
Son

test-fullautomation commented:
Hi Son,
very good!
Thank you,
 Thomas

test-fullautomation commented on README.rst:
```diff
@@ -21,31 +21,34 @@ Getting Started
+This RobotFramework_Testsuites package will support testing for many variants of product on the 
+same Robot project by switching between different configuration files via variant name.
 
 How to install
```
Hi Son,
reference is https://github.com/test-fullautomation/python-genpackagedoc
pip install is the first joice.
Please make "how to install" similar to genpackagedoc.
Thank you,
 Thomas

test-fullautomation replied:
Holger or I will do this in a final maintenance.

namsonx replied:
Hello Thomas,

Well noted!

Thank you,
Son

test-fullautomation commented on RobotFramework_Testsuites/Config/CConfig.py:
```diff
@@ -414,9 +416,20 @@ def updateCfg(sUpdateCfgFile):
+        try:
+            jsonDotdict = dotdictObj.dotdictConvert(oTmpJsonCfgData)
+            bDotdict = True
+        except:
```
Hi Son,
in order to make ergonomic and helpful error messages always catch the exception message and write something like:
```
logger.info(f"Could not convert json config to dotdict! Reason: '{msg}'")
```
Never use multiple exclamation mark. One is enough.

And why is this an info? We should exit here. Why should we process futher files when we have an error here?

Thank you,
 Thomas

namsonx replied:
Hello Thomas,

For your question "And why is this an info? We should exit here. Why should we process futher files when we have an error here?"
At the beginning during supporting BIST:RF or TA Framework, we recognized that some parameters in their configuration files contain "." in a parameter name. It makes dotdict converting process not successful, then we defined skip dotdict converting in this case.
I also mention this in our document: 
Note: In case a parameter name contains a ”.”, then it is not possible to use dotdict but the traditional way
${dict}[abc][def] is still working.

Thank you,
SOn

test-fullautomation replied:
Hi Son,
O.K. I got the point. Accepted...
Thank you for the explanation,
  Thomas

test-fullautomation commented on RobotFramework_Testsuites/Keywords/CSetup.py:
```diff
@@ -111,6 +117,8 @@ def testsuite_setup(self, sTestsuiteCfgFile=''):
         logger.info('Suite Path: %s' %(RobotFramework_Testsuites.CTestsuitesCfg.oConfig.sTestcasePath))
         logger.info('CfgFile Path: %s' %(RobotFramework_Testsuites.CTestsuitesCfg.oConfig.sTestCfgFile))
+        if RobotFramework_Testsuites.CTestsuitesCfg.oConfig.lLocalConfig != '':
+            logger.info('Local config file: %s' %(RobotFramework_Testsuites.CTestsuitesCfg.oConfig.lLocalConfig))
```
Hi Son,
long time ago I requested to use only f strings  instead of the old % method.
Than you,
 Thomas

namsonx replied:
Hello Thomas,

I will update this point, and I will create another pull-request to update all Robotframework-Testsuitesmanagement code.

Thank you,
Son

test-fullautomation commented on RobotFramework_Testsuites/Utils/LibListener.py:
```diff
@@ -80,7 +81,14 @@ def _start_suite(self, data, result):
+                    logger.error("local_config input must not be empty!!!")
+                else:
+                    RobotFramework_Testsuites.CTestsuitesCfg.oConfig.lLocalConfig = BuiltIn().get_variable_value('${LOCAL_CONFIG}').strip()
+            elif os.path.isfile(os.environ['ROBOT_LOCAL_CONFIG'] + "/local_config.json"):
```
Hi Son,
here we have a mißunderstanding.
```
elif os.path.isfile(os.environ['ROBOT_LOCAL_CONFIG']:
```
was what I mean. 
1. ```ROBOT_LOCAL_CONFIG``` is optional and by default not existing.
2. if variable ```ROBOT_LOCAL_CONFIG ``` is existing, then we load the content.
3. ```ROBOT_LOCAL_CONFIG``` has to have a fully valid filename.
4. We need to raise an error if ```ROBOT_LOCAL_CONFIG``` has a content, but ```isfile``` fails or the content is not valid.

Please add this update
https://github.com/test-fullautomation/robotframework-testsuitesmanagement/pull/45/files#diff-110f80967316b13e247ab4eb439fdfd960cb2ab8537722ff01bf0cd162e49941
according to our discussion.

Thank you,
 Thomas

namsonx replied:
Hello Thomas,

I will push the commit to modify work follow of local config feature as you mentioned here.

Thank you,
Son

test-fullautomation commented on atest/atest_component/local_config/testsuites/config/local_config_default_03.json:
```diff
@@ -0,0 +1,65 @@
+      "minorversion": "1",
+      "patchversion": "0"
+    },
+    "Maximum_version": "0.5.3",
```
Hi Son,
next version is 0.6.0.0
Please correct.
Thank you,
 Thomas

namsonx replied:
Hello Thomas,

Well noted!

Thank you,
Son

test-fullautomation commented on atest/atest_component/local_config/testsuites/config/local_config_default_03.json:
```diff
@@ -0,0 +1,65 @@
+      "patchversion": "0"
+    },
+    "Maximum_version": "0.5.3",
+    "Minimum_version": "0.4.10",
```
Hi Son,
Test will fail with 0.4.1.0. Miniumum is also 0.6.0.0
Thank you,
Thomas

namsonx replied:
Hello Thomas,

Well noted!

Thank you,
Son

test-fullautomation commented on atest/resources/atest_libs.py:
```diff
@@ -39,3 +45,18 @@ def subprocess_execution(testscript):
+    suite_dir = os.path.dirname(BuiltIn().get_variable_value("${SUITE_SOURCE}"))
+    os.chdir(suite_dir)
+    os.system("copy " + os.path.abspath("../../general_config/localconfig/local_config.json") \
+        + " " + os.environ['ROBOT_LOCAL_CONFIG'] + "\\local_config.json")
```
HI Son,
ROBOT_LOCAL_CONFIG must be inclusive filename.
Thank you,
 Thomas

namsonx replied:
Hello Thomas,

Well noted!

Thank you,
Son

test-fullautomation commented:
Hi Son,
please look into my findings and provide a new pull request.
Thank you,
 Thomas

test-fullautomation commented on README.md:
```diff
@@ -19,159 +19,38 @@ repository**
 
 ### How to install
 
 Firstly, clone **RobotFramework_Testsuites** repository to your machine
```
Hi Son,
this is unfortunately again wrong.
First proposal is to use ```pip install```. Please refer to [genpackagedoc readme](https://github.com/test-fullautomation/python-genpackagedoc#readme).
This is the reference. Best is to copy parts and adapt accordingly.
Thank you,
 Thomas

namsonx replied:
Hello Thomas,

I will update the document.

Thank you,
Son

HolQue commented:
Hello Son,

in the documentation you wrote:

"After installed RobotFramework AIO, the localconfig directory is created in:
Windows: C:\RobotTest\localconfig
Ubuntu: /home/<user>/RobotTest/localconfig"

and in the code I found: "os.environ['ROBOT_LOCAL_CONFIG'] + "/local_config.json""

This means: A new folder needs to be created and also a new environment variable. And the RobotFramework AIO installer does this - and not the setup.py of the robotframework-testsuitesmanagement repository? This implies a strong dependency between an open source repository and a company internal installer.

It would be helpful to have some more clarity about the amount of dependencies that are wanted or accepted here.

By the way: The availability of this environment variable ROBOT_LOCAL_CONFIG should be mentioned in the documentation - because this is useful to know.

HolQue reviewed (commented):
Hi Son,

seems that something goes wrong or is not clear to me. I tried to access a local configuration file with:

--variable local_config:"./localconfig/exercise-06_localconfig_bench1.json" "./exercise-06.robot"

But:
* The test prints the default value instead of the value defined in this local config
* No error message in case of file contains syntax errors
* No error message in case of file does not exist.
Looks like the command line variable "local_config" is ignored completely.

HolQue commented:
Hi Son,

regarding my previous hint about the local config not working. The file I used for testing you can find in

https://github.com/test-fullautomation/robotframework-tutorial/pull/16

exercise-06

HolQue commented:
Hi Son,

I got the latest update from your branch and installed again. Now the behavior is different.

When I try to access a local config file that does not exist, or when I try to access a local config file that contains syntax errors, then I get an error message "[ ERROR ] Loading of JSON configuration file failed! Reason: ..." and after this a warning "[ WARN ] Could not load local configuration file: ..."

I would say that is too much. When already an error message is given, then we do not need a warning telling the same.

When I create a new local config parameter, e.g.
"teststring_bench" : "I am the 'bench1' configuration of exercise 06"
then this is ignored completely. This value is not available during test execution.

But the example above is not what I want to do. I want to overwrite a parameter that is already initialized in one of the other configuration files.

Therefore I assumed that I have to do it in this way:

${params}['global']['teststring_bench'] : "I am the 'bench1' configuration of exercise 06"

But this causes troubles:

[ ERROR ] Loading of JSON configuration file failed! Reason: Could not set variable 'params['global']['teststring_bench']' with value 'I am the 'bench1' configuration of exercise 06'!
[ WARN ] Could not load local configuration file: ./localconfig/exercise-06_localconfig_bench1.json

What is the proper way to do this?

namsonx commented:
> local_config:"./localconfig/exercise-06_localconfig_bench1.json"

Hello Holger,

     I found the root cause come from the string value that is used to overridden.
     ${params}['global']['teststring_bench'] : "I am the 'bench1' configuration of exercise 06"
     The single quote inside the string value of Json configuration parameter is made this string value could not be assigned.
     I will check this conner case.

Thank you, 
Son

HolQue commented:
Hi Son,

another side effect: When executing other robot tests in command line I get this error message:

[ ERROR ] Calling method '_start_suite' of listener 'LibListener' failed: KeyError: 'ROBOT_LOCAL_CONFIG'

This deep dependency of the package to an environment variable that is not created by the package itself, is not nice.

And I wonder why a LibListener needs to observe the local config folder.

namsonx commented:
> Hi Son,
> 
> another side effect: When executing other robot tests in command line I get this error message:
> 
> [ ERROR ] Calling method '_start_suite' of listener 'LibListener' failed: KeyError: 'ROBOT_LOCAL_CONFIG'
> 
> This deep dependency of the package to an environment variable that is not created by the package itself, is not nice.
> 
> And I wonder why a LibListener needs to observe the local config folder.

Hello Holger,

For the environment variable 'ROBOT_LOCAL_CONFIG', I will update to the RobotFramework Testsuitesmangement documentation as we discussed.
For LibListener, I had designed using this LibListener of Robot Framework to hook the start_suite to load configuration object at the very first starting of robot run.

Thank you,
Son

HolQue commented:
Hi Son,

on first page of RobotFramework Testsuites.pdf the version is 0.2.2, but in history the latest version is 0.1.1

This should be aligned.

namsonx commented:
> Hi Son,
> 
> on first page of RobotFramework Testsuites.pdf the version is 0.2.2, but in history the latest version is 0.1.1
> 
> This should be aligned.

Hello Holger,

Actually, I do not know where the version in the first page of documentation came from :)
I just update the new version in History.tex file, in this case is 0.1.1, I don't know why & how 0.2.2 is added in the first page of generated document.
Do you have clue for this issue?

HolQue commented:
Hi Son,

the version information is taken out of the code automatically. In this package it's:

RobotFramework_Testsuites\packageversion.py

And this current version number should be aligned with the latest version on bottom of the history.

Because of the history also contains further informations that cannot be taken out of the code, we have in all packages these two positions where we have to maintain the version number.

namsonx commented:
Hello Holger, 

Thank you for your explanation! I have just updated new version in both History.tex and packageversion.py.

There is 1 issue related to packagedoc I observed when I try to add the picture '.\packagedoc\additional_docs\pictures\local_config_feature.png' the document.
I added this picture to Description.rst file by adding line ".. image:: ./pictures/local_config_feature.png", but the build failed with error message below:
LaTeX Warning: File `./pictures/local_config_feature.png' not found on input li
ne 254.


! Package pdftex.def Error: File `./pictures/local_config_feature.png' not foun
d: using draft setting.

See the pdftex.def package documentation for explanation.
Type  H <return>  for immediate help.
 ...

l.254 ...hics{./pictures/local_config_feature.png}

HolQue commented:
Hi Son,

in your branch the "picture" feature is commented out in packagedoc_config.json:
`# "PICTURES" : "./additional_docs/pictures",`
Please undo the comment.

Reason: As long as there are no pictures available the "pictures" folder is empty. Empty folders are ignored by GIT. But when the line "PICTURES" : "./additional_docs/pictures" is active, than GenPackageDoc throws an error in case of the folder is not existing. Therefore the feature is deactivated per default - and the user has to activate it in case he needs this feature.

HolQue commented:
Hi Son,

intermediately on my computer I removed all nested quotes from strings defined in JSON configuration files of tutorial exercise-06 and I went through the following scenarios

1 No variant name given in command line, no local configuration
2 Variant name given in command line, no local configuration
3 No variant name given in command line, with local configuration
4 Variant name given in command line, with local configuration

like described in this exercise.

With the following results:

1 : OK, got all default values
2 : Not OK; got default value instead of expected variant specific value
3 : OK, got bench specific value
4 : OK, got variant specific value and bench specific value

I have no idea what goes wrong in case 2.

Is it maybe possible for you to try to reproduce this use case? Would be helpful to detect if I made something wrong or something in the code needs to be fixed.

HolQue commented:
Hi Son,

I used a variant name that is not defined in the variants configuration file:

`--variable variant:"iamnotdefined"`

RobotFramework_Testsuites ignores this and I get the default values.

It would be better to throw an error, telling that the requested variant is not defined.

HolQue commented:
Hi Son,

when I provide a variant name in command line

--variable variant:"variant1"

but I miss to pass a variants configuration file to the Suite Setup

Suite Setup    testsuites.testsuite_setup

then RobotFramework_Testsuites ignores this.

But it should be possible to detect that no variant specific information is available. In this case it would make no sense to define a variant name in command line. Therefore in this case the RobotFramework_Testsuites should throw an error.

namsonx commented:
> : Not OK; got default value instead of expected variant specific value

Hello Holger,

For this scenario, we should discuss with Thomas before do change the current behavior.  Because as I remember we designed this behavior for level 2 configuration.

namsonx commented:
> Hi Son,
> 
> I used a variant name that is not defined in the variants configuration file:
> 
> `--variable variant:"iamnotdefined"`
> 
> RobotFramework_Testsuites ignores this and I get the default values.
> 
> It would be better to throw an error, telling that the requested variant is not defined.

Hello Holger,

For this scenario, we should discuss with Thomas before do change the current behavior.  Because as I remember we designed this behavior for level 2 configuration.

namsonx commented:
> Hi Son,
> 
> when I provide a variant name in command line
> 
> --variable variant:"variant1"
> 
> but I miss to pass a variants configuration file to the Suite Setup
> 
> Suite Setup testsuites.testsuite_setup
> 
> then RobotFramework_Testsuites ignores this.
> 
> But it should be possible to detect that no variant specific information is available. In this case it would make no sense to define a variant name in command line. Therefore in this case the RobotFramework_Testsuites should throw an error.

namsonx commented:
> Hi Son,
> 
> when I provide a variant name in command line
> 
> --variable variant:"variant1"
> 
> but I miss to pass a variants configuration file to the Suite Setup
> 
> Suite Setup testsuites.testsuite_setup
> 
> then RobotFramework_Testsuites ignores this.
> 
> But it should be possible to detect that no variant specific information is available. In this case it would make no sense to define a variant name in command line. Therefore in this case the RobotFramework_Testsuites should throw an error.

Hello Holger,

I was not aware of this case!
Yes, you're right, I will throw an error in case variant variable is set but there is no variants configuration file passed to the Suite Setup.
Thank you for your finding :)

HolQue commented:
Thinking about: 

We have 4 levels of loading a configuration. And we have an additional feature now: the local config.

This feature seems not to have an own level number. But this is unfavorable.

The local config can be used to overwrite already existing values. The outcome is: When a local config is involved,
we do not have the original configuration (indicated by level 1-4) any more, but a possibly modified one.

The level number should reflect this.

In command line I provided a local config:

--variable local_config:"./localconfig/exercise-06_localconfig_bench1.json"

And the log file indicates level 2 - that as per documentation belongs to a variant name provided in command line (or to the default variant):

Running with configuration level: 2
CfgFile Path: ./config/exercise-06_config_default.json
Local config file: ./localconfig/exercise-06_localconfig_bench1.json

But indicating level 2 is not correct, because we do not necessarily have the original variant specific set of values active during the test execution.

We also should not loose the information about the origin of the values: the variant specific configuration.

The outcome is in my opinion that the level number needs a modifier! Something like '2lc' (= level 2, modified by local config).

Or a little bit simpler: '2m' or '2+' (= level 2, modified).

HolQue commented:
Is it a requirement for configuration files to support Unicode characters?

I gave it a try:

${params}['global']['teststring_bench'] : "I am the bench1 SälfTest.ß.€.考.𠼭.𠼭 configuration of exercise 06"

But the special characters in the middle of the string are not displayed correctly in log file and also not in console:

teststring_bench : I am the bench1 SÃ¤lfTest.ÃŸ.â‚¬.è€ƒ.ð ¼­.ð ¼­ configuration of exercise 06

namsonx commented:
> Hi Son,
> 
> I used a variant name that is not defined in the variants configuration file:
> 
> `--variable variant:"iamnotdefined"`
> 
> RobotFramework_Testsuites ignores this and I get the default values.
> 
> It would be better to throw an error, telling that the requested variant is not defined.

Hello Holger,

I have just created the [pull-request](https://github.com/test-fullautomation/robotframework-testsuitesmanagement/pull/48) to improve TestsuitesManagement according to your comment.
You can do cherry-pick my commit to check again with your scenario: "git cherry-pick e49843bcc388aff06ac0c733111c2304fe5f7f99"

Thank you,
Son

namsonx commented:
> Thinking about:
> 
> We have 4 levels of loading a configuration. And we have an additional feature now: the local config.
> 
> This feature seems not to have an own level number. But this is unfavorable.
> 
> The local config can be used to overwrite already existing values. The outcome is: When a local config is involved, we do not have the original configuration (indicated by level 1-4) any more, but a possibly modified one.
> 
> The level number should reflect this.
> 
> In command line I provided a local config:
> 
> --variable local_config:"./localconfig/exercise-06_localconfig_bench1.json"
> 
> And the log file indicates level 2 - that as per documentation belongs to a variant name provided in command line (or to the default variant):
> 
> Running with configuration level: 2 CfgFile Path: ./config/exercise-06_config_default.json Local config file: ./localconfig/exercise-06_localconfig_bench1.json
> 
> But indicating level 2 is not correct, because we do not necessarily have the original variant specific set of values active during the test execution.
> 
> We also should not loose the information about the origin of the values: the variant specific configuration.
> 
> The outcome is in my opinion that the level number needs a modifier! Something like '2lc' (= level 2, modified by local config).
> 
> Or a little bit simpler: '2m' or '2+' (= level 2, modified).

Hello Holger,

Your proposal is nice, it helps user is aware of some parameters of Json configuration which loaded by level2 are modified.
Current log also indicates that local config is using:
Running with configuration level: 2
CfgFile Path: ./config/exercise-06_config_default.json
**Local config file: ./localconfig/exercise-06_localconfig_bench1.json**

The message **"Local config file: <path_to_local_config_file>"** only print out when local config feature is active.

namsonx commented:
> Is it a requirement for configuration files to support Unicode characters?
> 
> I gave it a try:
> 
> ${params}['global']['teststring_bench'] : "I am the bench1 SälfTest.ß.€.考.𠼭.𠼭 configuration of exercise 06"
> 
> But the special characters in the middle of the string are not displayed correctly in log file and also not in console:
> 
> teststring_bench : I am the bench1 SÃ¤lfTest.ÃŸ.â‚¬.è€ƒ.ð ¼­.ð ¼­ configuration of exercise 06

Hello Holger,

Unfortunately, it does not support Unicode characters yet, you can create a ticket for me to implement it in python-jsonpreprocessor package.

HolQue commented:
https://github.com/test-fullautomation/robotframework-testsuitesmanagement/pull/45#issuecomment-1340869639

Hi Son,

my concern is about the level number itself, and not about the additional entry "Local config file" in log file.

Why not both: The "Local config file" entry in log file AND a modified level number ?

test-fullautomation commented:
Yes, this is what we do in TML (I asked to review the TML Code as reference).
We log the Level + file, then if found and loaded the used local config file.
Then you know under which conditions exactly the test was executed.

test-fullautomation commented:
> > Hi Son,
> > I used a variant name that is not defined in the variants configuration file:
> > `--variable variant:"iamnotdefined"`
> > RobotFramework_Testsuites ignores this and I get the default values.
> > It would be better to throw an error, telling that the requested variant is not defined.
> 
> Hello Holger,
> 
> For this scenario, we should discuss with Thomas before do change the current behavior. Because as I remember we designed this behavior for level 2 configuration.

Hi Son,
Reference is again TML.
We throw an error and before we log all possible variants.

test-fullautomation commented:
> Thinking about:
> 
> We have 4 levels of loading a configuration. And we have an additional feature now: the local config.
> 
> This feature seems not to have an own level number. But this is unfavorable.
> 
> The local config can be used to overwrite already existing values. The outcome is: When a local config is involved, we do not have the original configuration (indicated by level 1-4) any more, but a possibly modified one.
> 
> The level number should reflect this.
> 
> In command line I provided a local config:
> 
> --variable local_config:"./localconfig/exercise-06_localconfig_bench1.json"
> 
> And the log file indicates level 2 - that as per documentation belongs to a variant name provided in command line (or to the default variant):
> 
> Running with configuration level: 2 CfgFile Path: ./config/exercise-06_config_default.json Local config file: ./localconfig/exercise-06_localconfig_bench1.json
> 
> But indicating level 2 is not correct, because we do not necessarily have the original variant specific set of values active during the test execution.
> 
> We also should not loose the information about the origin of the values: the variant specific configuration.
> 
> The outcome is in my opinion that the level number needs a modifier! Something like '2lc' (= level 2, modified by local config).
> 
> Or a little bit simpler: '2m' or '2+' (= level 2, modified).

Hi Holger,
Level2 basically should indicate that the path of the loaded config file is part of the robot file.
When we log after this, that we load a local config file, then we can assume that the orignially loaded file (by level2) is modified.
So, from my point of view no need to log "level 2, modified". This is redundent and even a bit error prone to implement.
Thank you,
Thomas

test-fullautomation reviewed (changes requested):
Hi Son,
looks like my review comments are not visible if I don't start a review.
Are the comments visible now?
Thank you,
 Thomas

namsonx commented:
> Hi Son,
> 
> looks like my review comments are not visible if I don't start a review.
> 
> Are the comments visible now?
> 
> Thank you,
> 
>  Thomas

Hello Thomas, 

Yeah, I can see your comments now 🙂 

Thank you,
Son

test-fullautomation commented:
> Hello Thomas,
> 
> Yeah, I can see your comments now 🙂
> 
> Thank you, Son

very good!
Then I mustn't forget to press the "Start Review" button with "request for changes" after my review. Good to know...

namsonx commented:
Hello Thomas,

I have just pushed the new commit base on your review comments.
Could you please help me check and give me your feedback if any?

Thank you,
Son

HolQue commented:
In the README files already the new term "RobotFramework_TestsuitesManagement" is used - mostly. But some position still contain the old one "RobotFramework_Testsuites package". This should be aligned.

HolQue commented:
Regarding:
"This RobotFramework_Testsuites package will support testing for many variants of product on the same Robot project by switching between different configuration files via variant name.

    Using configuration files in Json format
    Define 4 levels of configuration
    Local configuration
    Dotdict feature
"

(1) "via variant name" is not the only possibility. The other ones should also be mentioned.
(2) The item list after the first sentence contains also features. This is a strange mix. Either you explain all features in plain text or you list all features in form of a list of items.
(3) JSON is an abbreviation - and therefore should be written in capital letters completely.
(4) The item list contains rather short terms only and concentrates on technical details. Something like "Dotdict feature" is too much anonymous. This does not help the audience to understand. The question is: What is the idea behind this feature? What is the reason? This should be mentioned here at this early stage of documentation. That e.g. a feature is realized by a dotdict object is a technical detail that should not be placed in a readme. For such kind of detailed informations we have the PDF documentation.

HolQue commented:
"A detailed documentation of the Python Extensions Collection package can be found here: [RobotFramework_Testsuites.pdf](https://github.com/test-fullautomation/robotframework-testsuitesmanagement/blob/develop/RobotFramework_Testsuites/RobotFramework_Testsuites.pdf)"

Wrong package name: "Python Extensions Collection"

HolQue commented:
Working on the documentation of a component that is planned to be renamed, is rather unfavorable. Because too much text - and maybe also code - have to be adapted again after the renaming.

Therefore I suggest to do the renaming as soon as possible. And then continue with all other things that are necessary.

HolQue commented:
Regarding:
"RobotFramework AIO TESTSUITES MANAGEMENT DOCUMENTATION"

Please check if it is really required do have the term "AIO" inside. This readme is visible in GitHub and also in PyPi. No one knows there what "AIO" is.

Why is "TESTSUITES MANAGEMENT DOCUMENTATION" written in capital letters only? And usually "TESTSUITES" is written in two words: "test suites".

HolQue commented:
Even in case of a version mismatch all tests are executed. The result of all tests is "UNKNOWN" - for sure, but they are executed.

Why this? I thought that the version control mechanism is responsible for an interrupt of the execution in case of a version mismatch.

test-fullautomation commented on atest/atest_component/builtin_variable/testsuites/config/local_config.json:
```diff
@@ -0,0 +1,11 @@
+// Initially created by Pollerspoeck Thomas (CM-AI/PJ-CMD) / 30.10.2017
```
uups accidental copy&paste take over...

test-fullautomation reviewed (approved):
Hi Son,
for stabilization I merged now the latest state of #45.
Thank you,
 Thomas

test-fullautomation reviewed (approved):
approved...

test-fullautomation commented:
Hi Son,
I assume that no work on this branch is required anymore due to the fact that this is all part of #63.
Please confirm. Then I close this pull-request.
Thank you,
 Thomas

namsonx commented:
> Hi Son, I assume that no work on this branch is required anymore due to the fact that this is all part of #63. Please confirm. Then I close this pull-request. Thank you, Thomas

Hello Thomas,

Yes, I already resolved and merged all commits of pull-request #45, #59, #60, #61, and #62 to #63 
So you can abandon all these pull-requests

Thank you,
Son

test-fullautomation commented:
Closed. This is part of already merged #63.
